stream-chat-android-client/src/main/java/io/getstream/chat/android/client/errors/cause/StreamException.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.client.errors.cause
+
+public sealed class StreamException : Exception {
```

Comment:
Do we need both the StreamException and StreamCodeException as basically it seems like the same thing? The code exception doesn't contain the code at all so there's no distinction.

We could just have StreamException and a bunch of different cases, like the moderation deleted and moderation failed

Reply:
Also docs

Reply:
Wanted to keep separate backend-based exceptions and the ones we use internally inside the SDK only.
That's why I created 2 exceptions hierarchy

Reply:
While that makes sense, can't we just treat them as one exception set? They're both an exception and in our SDK, it shouldn't matter much if it's a BE one or ours, as it's all a "stream exception", something being wrong with data or our logic.

I'm just thinking that we're creating 2 sealed classes and we'll ahve to parse back and forth and these are public as well, so it's heavily increasing our API surface

Reply:
Ok, made a common parent `StreamException`:
StreamException:
    - StreamSdlException:
        - ...
    - StreamCodeException:
        - ...